test/client.test.ts

```diff
@@ -80,6 +80,75 @@ it("constructor throws if an invalid repository endpoint is provided", () => {
 	}).toThrowError(prismic.PrismicError);
 });
 
+it("constructor throws if a prismic.io endpoint is given that is not for Rest API V2", () => {
+	const fetch = vi.fn();
+
+	const originalNodeEnv = process.env.NODE_ENV;
+	process.env.NODE_ENV = "development";
+
+	expect(() => {
+		prismic.createClient("https://qwerty.cdn.prismic.io/api/v1", { fetch });
+	}).toThrowError(/only supports Prismic Rest API V2/i);
+	expect(() => {
+		prismic.createClient("https://qwerty.cdn.prismic.io/api/v1", { fetch });
+	}).toThrowError(prismic.PrismicError);
+
+	expect(() => {
+		prismic.createClient("https://example.com/custom/endpoint", { fetch });
+	}, "Non-prismic.io endpoints are not checked").not.toThrow();
+
+	expect(() => {
+		prismic.createClient("https://qwerty.cdn.prismic.io/api/v2", { fetch });
+	}, "A valid prismic.io V2 endpoint does not throw").not.toThrow();
+
+	expect(() => {
+		prismic.createClient(prismic.getRepositoryEndpoint("qwerty"), { fetch });
+	}, "An endpoint created with getRepositoryEndpoint does not throw").not.toThrow();
+
+	process.env.NODE_ENV === originalNodeEnv;
```

Comment:
Oops, there was a typo in the old test. This line should be:

```typescript
process.env.NODE_ENV = originalNodeEnv
```

Reply:
Nice catch!